src/test/java/bio/terra/service/snapshot/flight/create/SnapshotCreateFlightTest.java

```diff
@@ -100,4 +101,49 @@ void testSnapshotCreateFlight() {
         unlockDatasetStep.isSharedLock(),
         is(true));
   }
+
+  @Test
+  void testSnapshotCreateFlightByRequestId() {
+    UUID snapshotAccessRequestId = UUID.randomUUID();
+    SnapshotRequestModel request =
+        new SnapshotRequestModel()
+            .dataAccessControlGroups(DATA_ACCESS_CONTROL_GROUPS)
+            .addContentsItem(
+                new SnapshotRequestContentsModel()
+                    .mode(SnapshotRequestContentsModel.ModeEnum.BYREQUESTID)
+                    .requestIdSpec(
+                        new SnapshotRequestIdModel().snapshotRequestId(snapshotAccessRequestId)));
+    inputParameters.put(JobMapKeys.REQUEST.getKeyName(), request);
+    var flight = new SnapshotCreateFlight(inputParameters, context);
+
+    assertThat(
+        "Snapshot creation flight locks resources, then unlocks them, then writes response",
+        FlightTestUtils.getStepNames(flight),
+        containsInRelativeOrder(
```

Comment:
If this is all the steps, you can use `contains` here
```suggestion
        contains(
```